src/include/sof/audio/generic_processor/generic_processor.h

```diff
@@ -15,9 +15,19 @@ enum gp_state {
 	PP_STATE_RUN,
 };
 
+struct generic_processor_config {
+	uint32_t codec_id;
+	uint32_t reserved;
+	uint32_t sample_rate;
+	uint32_t sample_width;
+	uint32_t channels;
+};
+
 /* generic_processor private, runtime data */
 struct comp_data {
 	enum gp_state state;
+	struct generic_processor_config gp_config;
+
```

Comment:
Here we should go either with ``gp_`` prefix or ``generic_processor``. So, I think the structure should be named ``struct gp_config`` and then declare config member inside comp_data.

Reply:
Agree.